### PR TITLE
Use bounding box center as origin when limiting by distance

### PR DESCRIPTION
**Pull request by mporkola**
The point returned by Google was not the center, which caused some weird behaviour when searching with distance limit.

### CODE REVIEW

File: app/controllers/homepage_controller.rb [hunk 1]
```diff
@@ -297,6 +298,26 @@ def search_coordinates(latlng)
...
   end
 
+  def bounding_box_center(corners)
+    lat1, lng1, lat2, lng2 = corners.split(',').map { |n| toRadians(n) }
+    bx = Math.cos(lat2) * Math.cos(lng2 - lng1)
+    by = Math.cos(lat2) * Math.sin(lng2 - lng1)
+    lat_c = Math.atan2(Math.sin(lat1) + Math.sin(lat2),
+                        Math.sqrt((Math.cos(lat1)+bx)*(Math.cos(lat1)+bx)+by*by))
+    lng_c = lat1 + Math.atan2(by, Math.cos(lng1) + bx)
+    { latitude: toDegrees(lat_c), longitude: toDegrees(lng_c) }
```
Comment (mporkola) [inline app/controllers/homepage_controller.rb]: I'll document this algorithm somehow

Comment (Gnito) [inline app/controllers/homepage_controller.rb]: @mporkola, yesterday, I tried to [link] (http://stackoverflow.com/questions/29322582/how-to-calculate-center-of-bounds-without-googlemaps) you this solution. Sorry, I wasn't more explicit.
```
var center = (new google.maps.LatLngBounds(bounds.southeast, bounds.northeast)).getCenter();
```

File: app/controllers/homepage_controller.rb [hunk 2]
```diff
@@ -297,6 +298,30 @@ def search_coordinates(latlng)
...
+    unless (lat1.present? && lng1.present? && lat2.present? && lng2.present?)
+      ArgumentError.new("Format of corners \"#{corners}\" wasn't \"lat1,lng1,lat2,lng2\" ")
+    end
+    bx = Math.cos(lat2) * Math.cos(lng2 - lng1)
+    by = Math.cos(lat2) * Math.sin(lng2 - lng1)
+    lat_c = Math.atan2(Math.sin(lat1) + Math.sin(lat2),
+                       Math.sqrt((Math.cos(lat1) + bx) * (Math.cos(lat1) + bx) + by * by))
+    lng_c = lat1 + Math.atan2(by, Math.cos(lng1) + bx)
+    { latitude: toDegrees(lat_c), longitude: toDegrees(lng_c) }
+  end
```
Comment (mporkola) [inline app/controllers/homepage_controller.rb]: "Midpoint" on this page http://www.movable-type.co.uk/scripts/latlong.html

File: app/controllers/homepage_controller.rb [hunk 3]
```diff
@@ -297,6 +298,30 @@ def search_coordinates(latlng)
...
+
+  def toRadians(deg)
```
Comment (rap1ds) [inline app/controllers/homepage_controller.rb]: Should be snake_case

**Comment by rap1ds** [comment]
Maybe this piece of code could be moved to some helper method? E.g. LatLngUtils or GeoLocationUtils or something like that. HomepageController is already quite big.

**Comment by rap1ds** [comment]
LGMT

**Pull request merged**